Replicator/PropertyEncryption_stub.cc

```diff
@@ -0,0 +1,77 @@
+//
+// PropertyEncryption_stub.cc
+//
+// Copyright (C) 2020 Jens Alfke. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "PropertyEncryption.hh"
+
+#ifdef COUCHBASE_ENTERPRISE
+    // NOTE: PropertyEncryption.cc is not in this repo, and is not open source.
+    // It is part of Couchbase Lite Enterprise Edition (EE), which can be licensed in binary form
+    // from Couchbase.
+    #include "../../../couchbase-lite-core-EE/Replicator/PropertyEncryption.cc"
+#else
+
+// Stubs for CE:
+
+#include "Error.hh"
+#include "fleece/Mutable.hh"
+
+namespace litecore::repl {
+    using namespace fleece;
+
+    fleece::MutableDict EncryptDocumentProperties(fleece::slice docID,
+                                                  fleece::Dict doc,
+                                                  C4ReplicatorPropertyEncryptionCallback callback,
+                                                  void *callbackContext,
+                                                  C4Error *outError) noexcept
+    {
+        // In CE, prevent any encryptable property from being accidentally pushed.
+        // This may happen if a database was created and used with EE, and sensitive data added,
+        // and then it's opened with a CE implementation.
+        if (outError)
+            *outError = {};
+        for (DeepIterator i(doc); i; ++i) {
+            if (i.key() == C4Document::kObjectTypeProperty) {
+                if (i.value().asString() == C4Document::kObjectType_Encryptable) {
```

Comment:
I believe this needs to check for the presence of an unencrypted value.  If there is no unencrypted value present then it can be sent normally (perhaps by returning the same value that was passed in if we are going to consider a null encryptedRoot to be invalid as suggested above?).

Reply:
I don't understand this comment. Checking for an unencrypted value is exactly what this function does. If there is none it returns null. It can't return the value passed in because it takes an immutable value and returns a mutable one.

Reply:
I think @borrrden means if the value is already encrypted (containing kC4EncryptedCiphertextProperty instead of kC4EncryptableValueProperty), it is not an error (OK to push).

Reply:
That seemed reasonable, but when I looked at the code I realized that this never occurs. When the property is encrypted it's changed into the server schema, and vice versa. So there is never an `"@type":"EncryptedProperty"` object with a `ciphertext` property -- in fact the constant `kC4EncryptedCiphertextProperty` is never used anywhere and could be removed.

Reply:
Oh, yeah, that does seem to be the case!